src/Workspaces/Core/Portable/FindSymbols/SymbolFinder_Declarations_CustomQueries.cs

```diff
@@ -100,7 +101,7 @@ public static async Task<IEnumerable<ISymbol>> FindSourceDeclarationsAsync(Proje
 
             using (Logger.LogBlock(FunctionId.SymbolFinder_Project_Predicate_FindSourceDeclarationsAsync, cancellationToken))
             {
-                if (await project.ContainsSymbolsWithNameAsync(query.GetPredicate(), filter, cancellationToken).ConfigureAwait(false))
+                if (project.State.SourceGenerators.Any() || await project.ContainsSymbolsWithNameAsync(query.GetPredicate(), filter, cancellationToken).ConfigureAwait(false))
```

Comment:
effectively, this is the same as pushing the boolean check into ContainsSymbolsWithNameAsync and basically having that function go "whelp... if htere are source generators, guess i can't do anything smart/fast".  We need to *not* go down that path.  This may require some sort of smarts on behalf of SGs to make it possible to have efficient features along with SGs. But we can't lose efficiency just when an SG is present.

Tagging @jasonmalinowski as this keeps coming up and is very concerning to me :(